src/inc_xram.a51

```diff
@@ -0,0 +1,80 @@
+;    Copyright (c) 2007, 2008 Florian MAZEN and Pierre COL
+;    
+;    This program is free software: you can redistribute it and/or modify
+;    it under the terms of the GNU General Public License as published by
+;    the Free Software Foundation, either version 3 of the License, or
+;    (at your option) any later version.
+;
+;    This program is distributed in the hope that it will be useful,
+;    but WITHOUT ANY WARRANTY; without even the implied warranty of
+;    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;    GNU General Public License for more details.
+;
+;    You should have received a copy of the GNU General Public License
+;    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+; Segmentation of external memory now moved to an own file so memory structure can be seen easily
+; and DPTR can be loaded with 16 Bit values, also
+
+
+XSEG	AT 0000H						; RAM_Area_Config, don't change the order of variables					
+
+RAM_ID_CODE:		DS	1
+RAM_CONFIG_SUM:		DS	1
+RAM_FREQ_SUM:		DS	1
+RAM_SHIFT_SUM:		DS	1
+RAM_STATE_SUM:		DS	1
+
+XSEG	AT 0010H						; why this offset?				
+RAM_CHAN:			DS	1
+RAM_MODE:			DS	1
+RAM_SQUELCH:		DS	1
+RAM_MAX_CHAN:		DS	1
+RAM_SHIFT_HI:		DS	1
+RAM_SHIFT_LO:		DS	1
+RAM_PLL_DIV_HI:		DS	1
+RAM_PLL_DIV_LO:		DS	1
+RAM_SCAN_DURATION:	DS	1
+
+;RAM_ID_CODE		EQU	000h
+;RAM_CONFIG_SUM		EQU	001h
+;RAM_FREQ_SUM		EQU	002h
+;RAM_STATE_SUM		EQU	003h
+;RAM_CHAN			EQU	010h
+;RAM_MODE			EQU	011h
+;RAM_SQUELCH		EQU	012h
+;RAM_MAX_CHAN		EQU	013h
+;RAM_SHIFT_HI		EQU	014h
+;RAM_SHIFT_LO		EQU	015h
+;RAM_PLL_DIV_HI		EQU	016h
+;RAM_PLL_DIV_LO		EQU	017h
+;RAM_SCAN_DURATION	EQU	018h
+
+ID_CODE			EQU	040h
+
+XSEG	AT 0100H						; RAM_AREA_FREQ				
+Ch0_RX_freq:		DS 2				; 2 Bytes per Channel
+Ch1_RX_freq:		DS 2				; reserved for 128 channels max
+Chx_RX_freq: 		DS 252				
```

Comment:
ok now I guess it's a kind of programming style.
I've never defined ram-, code- or bitvariables via equations. E.g. its hard to insert variable later on...
I understand why you did this for the channel config segments (256 bytes, each), I don't understand why this was done for standard variable and bit definitions. 
But you are the master of this code, if you want that I change it I will do so

Reply:
Readin your reply, I'm not sure if i was clear :)
My question is why declaring theses 3 lines instead of one Ch0_RX_freq:		DS 256 ?
I have not use this instruction before, perhaps you have technical issue

Reply:
ok, now I understood the reason of your question. 
I wanted to do calculations also with the "Ch1_RX_freq" variable, but I didn't need this later anymore.
That means I will change the code

Reply:
changed